distributed_shampoo/fsdp_shampoo.py

```diff
@@ -456,12 +524,34 @@ def _compute_and_log_root_inverse_residuals(
             f"Relative Residual (||X_hat^-r - A||_inf / ||A||_inf) Average: {torch.mean(relative_residuals)}, Quantiles [0, 25, 50, 75, 100]: {torch.quantile(relative_residuals, quantiles, interpolation='nearest')}"
         )
 
+    @torch.no_grad()
+    def _apply_weight_decay(self):
+        """Incorporate weight decay into the gradient if we are not using decoupled weight decay.
+
+        Equivalent to adding an L2-regularization term:
+          F(w) + lambda * ||w||^2.
+
+        """
+        for group in self.param_groups:
+            weight_decay = group[WEIGHT_DECAY]
+            for p in group[PARAMS]:
+                # skip parameters not on worker
+                if p.numel() == 0:
+                    continue
+
+                grad = p.grad
+                if grad is None:
+                    continue
```

Comment:
Let's combine these into a single line check.
```suggestion
                if p.numel() == 0 or p.grad is None:
                    continue
                grad = p.grad
```